lib/device.h

```diff
@@ -30,9 +30,10 @@ class UDEVNOTIFIERSHARED_EXPORT Device
     QString node() const;
     QString subsystem() const;
     QString type() const;
+    DevicePrivate * const d;
```

Comment:
do **NOT** move the private pointer into the `public` area. This is **NOT** how you use d-pointers. If you want to access its members, do so via functions. If any needed functions are missing, add them to the base class